### PR TITLE
roll: extract shouldStop function and test it

### PR DESCRIPTION
**Pull request by petertseng**

### CODE REVIEW

**Comment by petertseng** [comment]
foot note is that I'm not 100% sure this makes our lives better, but I made enough of a mistake in this logic the first time that I think I'm motivated to try to get it right.

Obviously a test for `Roll` would still be great. Still need to investigate how to make that happen.

File: pkg/roll/run_update.go [hunk 1]
```diff
@@ -73,6 +73,15 @@ type Update interface {
 	Run(quit <-chan struct{}) bool
 }
 
+// returned by shouldStop
+type deployAction int
```
Comment (anthonybishopric) [inline pkg/roll/run_update.go]: nit `ruStep` maybe?

Comment (petertseng) [inline pkg/roll/run_update.go]: I'll take it! commit has now been amended.

**Comment by petertseng** [comment]
limited scope of `nextAction` to the `if` block

**Comment by mpuncel** [comment]
I like this

**Comment by mpuncel** [comment]
:+1:

**Pull request merged**